utils/apollo/schema/resolvers.js

```diff
@@ -1,54 +1,130 @@
-      // Query by Name and Email
+      // Fetch only the current session user by Name and Email
       if (email && name) {
         try {
           const user = await User.findOne({ name, email });
-          data.push(user);
+          return [user];
         } catch (err) {
-          console.debug(err);
+          console.debug('Cannot fetch user', err);
         }
-        return data;
       }
 
       // For Query by ID
-      for (let id of userIds) {
-        try {
-          const user = await User.findById(id);
-          data.push(user);
-        } catch (err) {
-          console.debug(err);
-        }
+      const filter = {
+        _id: {
+          $in: userIds,
+        },
+      };
+
+      let users = [];
+      try {
+        users = await User.find(filter);
+        console.debug('users', users);
+      } catch (err) {
+        console.debug(err);
       }
-      return data;
+
+      return users;
     },
 
     spaces: async (parent, args, context) => {
       const { spaceIds } = args;
       console.debug('args', args, 'spaceIds', spaceIds);
-      const data = [];
 
-      await dbConnect();
+      // Fetch all spaces
+      if (spaceIds?.length == 0) {
+        const spaces = await Space.find({}).populate('participants').exec();
+        return spaces;
+      }
 
-      for (let spaceId of spaceIds) {
-        try {
-          const space = await Space.findOne({ spaceId });
-          data.push(space);
-        } catch (err) {
-          console.debug(err);
-        }
+      // For Query by ID
+      const filter = {
+        spaceId: {
+          $in: spaceIds,
+        },
+      };
+
+      let spaces = {};
+      try {
+        spaces = await Space.find(filter).populate('participants').exec();
+      } catch (err) {
+        console.debug(err);
+      }
+      return spaces;
+    },
+  },
+  Mutation: {
+    createSpace: async (parent, args, context) => {
+      console.debug('createSpace', args);
+      const { input } = args;
+      const { name, description, userId, spaceId } = input;
+
+      let result = {};
+      try {
+        const space = new Space({
+          name,
+          description,
+          participants: [],
+          spaceId,
+          isActive: true,
+          music: '',
+        });
+
+        result = await space.save();
+        console.debug('new space', result);
+      } catch (err) {
+        console.debug('Cannot upload new space to database: ', err);
+      }
+      return result;
+    },
+    addUserToSpace: async (parent, args, context) => {
+      console.debug('AddUserToSpace', args);
+      const {
+        input: { spaceId, userId },
+      } = args;
+
+      let space = {};
+      try {
+        // The new option returns the space after the update
+        const filter = { spaceId };
+        const update = {
+          $push: { participants: userId },
+        };
+        space = await Space.findOneAndUpdate(filter, update, { new: true }).populate('participants').exec();
```

Comment:
why do we want to add `{ new: true }`? This would be a good place to add a comment to explain why you added that as an option

Reply:
For the new:true option, MongoDB returns the object before it was updated by default, so setting new:true returns the object after it was updated instead.